lib/manageiq/floe/workflow/state.rb

```diff
@@ -42,9 +42,13 @@ def context
         def run!(input)
           logger.info("Running state: [#{name}] with input [#{input}]")
 
+          output = block_given? ? yield : input
```

Comment:
If this returns `output, next_state`, then I think you can do `choice` here too.  Perhaps other methods would return nil fir next_state, and then the next line can be `next_state ||=` ?  Thoughts?

Reply:
:+1: Done